pkg/workloadmeta/collectors/internal/remote/process_collector/process_collector_test.go

```diff
@@ -240,9 +236,8 @@ func TestCollection(t *testing.T) {
 			mockConfig.Set("language_detection.enabled", true)
 			// remote process collector server (process agent)
 			server := &mockServer{
-				responses:       test.serverResponses,
-				errorResponse:   test.errorResponse,
-				currentResponse: 0,
+				responses:     test.serverResponses,
+				errorResponse: test.errorResponse,
 			}
 			grpcServer := grpc.NewServer()
```

Comment:
This `grpcServer` is started by `grpc.Serve(lis)` in a goroutine spawned line 248.
But I couldn’t find where `grpcServer.Stop()` is called.
So, I feel that the goroutine is leaked.

Reply:
Nice catch I'll fix that